### PR TITLE
Using wp_set_option_autoload() for theme options

### PR DESCRIPTION
**Pull request by Rajinsharwar**
Using wp_set_option_autoload() to set no to the old theme's autoload option.

Trac ticket: https://core.trac.wordpress.org/ticket/39537

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/theme.php [hunk 1]
```diff
@@ -866,6 +866,13 @@ function switch_theme( $stylesheet ) {
 
 	update_option( 'theme_switched', $old_theme->get_stylesheet() );
 
+	// Set autoload=no for the previous theme
+	$old_theme_stylesheet = $old_theme->get_stylesheet();
+	wp_set_option_autoload( "theme_mods_$old_theme_stylesheet", 'no' );
```
Comment (mukeshpanchal27) [inline src/wp-includes/theme.php:871]: This adjustment alters the autoload option for the previous theme. If the user had activated multiple themes previously, it becomes essential to verify and modify the autoload value for those themes as well. What do you think?

Comment (felixarntz) [reply src/wp-includes/theme.php:871]: @mukeshpanchal27 I think this is only an issue for existing sites that had multiple themes active before updating to WordPress 6.5. So I think this is better addressed separately via your new ticket https://core.trac.wordpress.org/ticket/59975 and #5711.

Any site that starts with WordPress 6.5 should not have this problem as every time the theme is switched the previous theme's theme mods are set to not autoload. So there should at no point be more than one theme with autoloaded theme mods.

Comment (felixarntz) [inline src/wp-includes/theme.php:871]: We need to also cover a potential parent theme here (i.e. if the old `template` is different from the old `stylesheet`). We could then use `wp_set_options_autoload()` to reduce the database requests needed.
```suggestion
	$old_theme_mods = array_unique(
		array(
			"theme_mods_{$old_theme->get_stylesheet()}",
			"theme_mods_{$old_theme->get_template()}"
		)
	);
	wp_set_options_autoload( $old_theme_mods, 'no' );
```

File: src/wp-includes/theme.php [hunk 2]
```diff
@@ -866,6 +866,13 @@ function switch_theme( $stylesheet ) {
 
 	update_option( 'theme_switched', $old_theme->get_stylesheet() );
 
+	// Set autoload=no for the previous theme
```
Comment (felixarntz) [inline src/wp-includes/theme.php:869]: Missing period after inline comment.
```suggestion
	// Set autoload=no for the previous theme.
```

File: src/wp-includes/theme.php [hunk 3]
```diff
@@ -866,6 +866,13 @@ function switch_theme( $stylesheet ) {
...
+	$old_theme_stylesheet = $old_theme->get_stylesheet();
+	wp_set_option_autoload( "theme_mods_$old_theme_stylesheet", 'no' );
+
+	// Set autoload=yes for the switched theme if not already set
+	wp_set_option_autoload( "theme_mods_$stylesheet", 'yes' );
```
Comment (felixarntz) [inline src/wp-includes/theme.php:874]: Same here:
```suggestion
	// Set autoload=yes for the switched to theme if not already set.
	$new_theme_mods = array_unique(
		array(
			"theme_mods_{$stylesheet}",
			"theme_mods_{$template}"
		)
	);
	wp_set_options_autoload( $new_theme_mods, 'yes' );
```

**Review by felixarntz — COMMENTED** [review]
Thank you for the PR @Rajinsharwar, I left a few points of feedback.

Would you also be able to add a unit test for the new functionality in `switch_theme()`?

**Comment by felixarntz** [comment]
Closing in favor of #5706. Thanks for working on this!

**Pull request closed without merging**